packages/theme-ui/src/custom-properties.js

```diff
@@ -0,0 +1,97 @@
+// convert theme and style objects to CSS custom properties
+import { css } from '@styled-system/css'
+
+const toVarName = key => `--theme-ui-${key}`
+const toVarValue = (key, value) => `var(${toVarName(key)}, ${value})`
+
+const join = (...args) => args.filter(Boolean).join('-')
+
+const numberScales = {
+  fontWeights: true,
+  lineHeights: true,
+}
+const reservedKeys = {
+  useCustomProperties: true,
+  initialColorMode: true,
+}
+
+const toPixel = (key, value) => {
+  if (typeof value !== 'number') return value
+  if (numberScales[key]) return value
+  return value + 'px'
+}
+
+// convert theme values to custom properties
+export const toCustomProperties = (obj, parent, themeKey) => {
```

Comment:
This *should* work for the entire theme object, and could potentially be replaced by the work in `@theme-ui/custom-properties` in the future